code/game/machinery/OpTable.dm

```diff
@@ -43,60 +40,58 @@
 		return TRUE
 
 /obj/machinery/optable/CanPass(atom/movable/mover, turf/target, height=0)
-	if(height==0) return 1
-
+	if(height==0)
+		return TRUE
 	if(istype(mover) && mover.checkpass(PASSTABLE))
-		return 1
+		return TRUE
 	else
-		return 0
+		return FALSE
 
 
-/obj/machinery/optable/MouseDrop_T(atom/movable/O as mob|obj, mob/user as mob)
-	if(usr.stat || (!ishuman(user) && !isrobot(user)) || user.restrained() || !check_table(user) || user.IsWeakened() || user.stunned)
+/obj/machinery/optable/MouseDrop_T(atom/movable/O, mob/user)
+	if(!ishuman(user) && !isrobot(user)) //Only Humanoids and Cyborgs can put things on this table
 		return
-
-	if(!ismob(O)) //humans only
+	if(!check_table()) //If the Operating Table is occupied, you cannot put someone else on it
 		return
-
-	if(istype(O, /mob/living/simple_animal) || istype(O, /mob/living/silicon)) //animals and robots dont fit
+	if(user.restrained() || user.buckled || user.IsWeakened() || user.stunned || user.incapacitated()) //Is the person trying to use the table incapacitated or restrained?
 		return
+	var/mob/living/carbon/C = O
+	if(!C) //Only Carbon mobs can go on this table, if C is null, get out of here
+		return
+	take_patient(C, user)
```

Comment:
```suggestion
	take_patient(O, user)
```
Don't have to cast it here. The check doesn't do much either as you can do this:
```
var/datum/t1/T = new()
var/datum/t2/T2 = T
if(T2)
 // will be true since T2 is assigned (just not the same type)
```

Reply:
This doesn't exclude `isanimal` and `isrobot`

Reply:
```
	if(isanimal(O) && isrobot(O)) //Only Carbon mobs can go on this table, if C is null, get out of here
		return
	take_patient(O, user)
```

Reply:
Changed to:
```
if(!iscarbon(O) || !ismob(O))
```
ismob() is needed to ensure that we're not placing an object on the table. If we do, it creates a runtime error in `take_patient()`, so we need the return. The ismob() check was removed based on a previous suggestion - the comments next to the original line of code led us to believe it did something else. Not sure where my head has been lately in terms of understanding things, apologies.

iscarbon() check replacing the casting since the casting wasn't actually helping check as you said.